UnityProject/Assets/Scripts/Health/Objects/Integrity.cs

```diff
@@ -246,6 +249,8 @@ public override void OnStartClient()
 
 			CheckDestruction(explodeOnDestroy);
 
+			OnDamaged?.Invoke();
```

Comment:
Damage event already exists:

OnApplyDamage

Reply:
you can't add listeners to it

Reply:
You can, that is why it exists. Thats what the function you use:

```private void OnDamageReceived(DamageInfo damageInfo) { }```